### PR TITLE
OCM-7049 | fix: allow proxy to be reset

### PR DESCRIPTION
**Pull request by gdbranco**
[OCM-7049](https://issues.redhat.com//browse/OCM-7049)

**Change type**
- [ ] New feature
- [x] Bug fix
- [ ] Build
- [ ] CI
- [] Documentation
- [ ] Performance
- [ ] Refactor
- [ ] Style
- [ ] Unit tests
- [x] Subsystem tests

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.

### CODE REVIEW

File: provider/clusterrosa/classic/cluster_rosa_classic_resource.go [hunk 1]
```diff
@@ -1542,29 +1541,32 @@ func populateRosaClassicClusterState(ctx context.Context, object *cmv1.Cluster,
 			state.Proxy = &proxy.Proxy{}
 		}
 		httpProxy, ok := proxyObj.GetHTTPProxy()
-		if ok {
+		if ok && httpProxy != "" {
```
Comment (nirarg) [inline provider/clusterrosa/classic/cluster_rosa_classic_resource.go:1544]: Why using empty string for reset the value and not using null for this?
Before this change, what happen if user update the value from valid value to null?
`httpProxy = null`

Comment (gdbranco) [reply provider/clusterrosa/classic/cluster_rosa_classic_resource.go:1544]: Nothing happened, plus the API usually omits null. It currently expects empty strings to reset

File: provider/cluster/cluster_resource.go [hunk 2]
```diff
@@ -203,7 +202,6 @@ func (r *ClusterResource) Schema(ctx context.Context, req resource.SchemaRequest
 				Description: "proxy",
 				Attributes:  proxy.ProxyResource(),
 				Optional:    true,
-				Validators:  []validator.Object{proxy.ProxyValidator()},
```
Comment (nirarg) [inline provider/cluster/cluster_resource.go:206]: I suggest to add this to the description - empty string used to reset

**Pull request merged**